### PR TITLE
Modernise pom

### PR DESCRIPTION
**Pull request by timja**
### What has been done
1. Modernise pom by updating to newer versions of dependencies and required Jenkins version
	- Jenkins plugins 2.32 -> 4.7
	- Jenkins 2.0 -> 2.204.6
2. Update required Java version 7 -> 8
3. Use https://github.com/jenkinsci/bom to satisfy upper boundary dependencies and no longer manage explicit versions but rather rely on 

### How to test
1. Make sure project compiles and CI is 🟢
2. Perform the plugin update to new version packaged from current PR to smoke test plotting capabilities after migration to Java 8
3. Make sure Jenkins log doesn't throw any new errors/warnings

### Checklist

- [ ] Git commits follow [best practices](https://chris.beams.io/posts/git-commit/) 
- [ ] Build passes in Jenkins 
- [ ] Appropriate tests or explanation to why this change has no tests 
- [x] Pull Request is marked with appropriate label (see `.github/release-drafter.yml`) 
- [ ] JIRA issue is well described (problem explanation, steps to reproduce, screenshots) 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by vgaidarji — COMMENTED** [review]
Changes in the PR are good with me 👍 
No need to update any classes yet to use Java 8 features.

How does this help with https://issues.jenkins-ci.org/browse/JENKINS-62800? Or it's just the initial project cleanup and the actual fix comes next?

Please provide before/after screenshots for the `tables to div` fix when ready.

**Comment by vgaidarji** [comment]
image
The build is failing due to Checkstyle warnings. It wasn't the case before and should be caused by upstream changes in [`buildPlugin()`](https://github.com/jenkinsci/plot-plugin/blob/7e386242d00a2103252b5b26e2856cbb272640f5/Jenkinsfile#L1)

Those `TODO`'s should probably be removed.

**Comment by vgaidarji** [comment]
Since [buildPlugin()](https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy) already works with [JDK 8 by default](https://github.com/jenkins-infra/pipeline-library/blob/284733e204a0028133a158027e6fed002d6ab73e/vars/buildPlugin.groovy#L283), no changes required
image

File: pom.xml [hunk 1]
```diff
@@ -4,17 +4,28 @@
...
-    <!-- Java Level to use. Java 7 required when using core >= 1.612 -->
-    <java.level>7</java.level>
+    <jenkins.version>2.204.6</jenkins.version>
+    <java.level>8</java.level>
   </properties>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.jenkins.tools.bom</groupId>
```
[the remark this answers is not in the document]
Comment (timja) [reply pom.xml:20]: In general you shouldn’t need to update the deps just update the bom version

**Comment by timja** [comment]
This was just an initial PR to make testing the plugin easier against the new Jenkins version

File: pom.xml [hunk 2]
```diff
@@ -4,17 +4,28 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.32</version>
+    <version>4.6</version>
```
Comment (timja) [inline pom.xml:7]:
```suggestion
    <version>4.7</version>
```

File: pom.xml [hunk 3]
```diff
@@ -4,17 +4,28 @@
...
-    <jenkins.version>2.0</jenkins.version>
-    <!-- Java Level to use. Java 7 required when using core >= 1.612 -->
-    <java.level>7</java.level>
+    <jenkins.version>2.204.6</jenkins.version>
+    <java.level>8</java.level>
   </properties>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
```
Comment (timja) [inline pom.xml:19]:
```suggestion
      <!-- https://github.com/jenkinsci/bom -->
      <dependency>
```

File: pom.xml [hunk 4]
```diff
@@ -173,25 +130,5 @@
     <url>https://github.com/jenkinsci/plot-plugin</url>
     <tag>HEAD</tag>
   </scm>
-
-  <distributionManagement>
```
[the remark this answers is not in the document]
Comment (timja) [reply pom.xml:177]: they are in the parent pom, haven't been needed here for ages
https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml#L856

**Comment by vgaidarji** [comment]
@timja thanks for updates here. Will be ok updating the current PR to be focused only on Java 8 migration and adjustments to `pom.xml`? So that we can merge this PR and it doesn't stay open for long time. And then separate PR to be opened focused only issues.jenkins-ci.org/browse/JENKINS-62800?
If you agree, can you please adjust the PR title/description to match that and we merge it.

**Comment by timja** [comment]
> @timja thanks for updates here. Will be ok updating the current PR to be focused only on Java 8 migration and adjustments to `pom.xml`? So that we can merge this PR and it doesn't stay open for long time. And then separate PR to be opened focused only issues.jenkins-ci.org/browse/JENKINS-62800?
> If you agree, can you please adjust the PR title/description to match that and we merge it.

done

**Comment by vgaidarji** [comment]
@timja I've updated the PR description and was about to do final testing before merge but CI is not passing and seems like there's some issue with update to 4.7 version of Jenkins plugins performed here https://github.com/jenkinsci/plot-plugin/pull/63#discussion_r491827713.
image

I have a custom step for code coverage calculation and upload in [Jenkinsfile](https://github.com/jenkinsci/plot-plugin/blob/e47694578784ddf760a23e19393da6b19260abb7/Jenkinsfile#L6) and it fails to find `org.jenkins-ci.plugins:plugin:pom:4.7` dependency.
```
15:49:01  [ERROR]     Non-resolvable parent POM for org.jenkins-ci.plugins:plot:2.1.9-SNAPSHOT: 
Could not find artifact org.jenkins-ci.plugins:plugin:pom:4.7 in central (https://repo.maven.apache.org/maven2) 
and 'parent.relativePath' points at no local POM @ line 4, column 11 -> [Help 2]
```

**Comment by timja** [comment]
probably would be easier to just change to buildPlugin? it has code coverage support built in, using code-coverage-api plugin rather than codecov

File: pom.xml [hunk 5]
```diff
@@ -4,17 +4,29 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.32</version>
+    <version>4.15</version>
```
Comment (timja) [inline pom.xml:7]:
```suggestion
    <version>4.16</version>
```

File: pom.xml [hunk 6]
```diff
@@ -4,17 +4,29 @@
...
+    <java.level>8</java.level>
   </properties>
 
+  <dependencyManagement>
+    <dependencies>
+      <!-- https://github.com/jenkinsci/bom -->
+      <dependency>
+        <groupId>io.jenkins.tools.bom</groupId>
+        <artifactId>bom-2.222.x</artifactId>
+        <version>24</version>
```
Comment (timja) [inline pom.xml:23]:
```suggestion
        <version>25</version>
```

**Pull request merged**